### PR TITLE
Use intra-doc links in `core::future::future` and `core::num::dec2flt`

### PR DESCRIPTION
**Pull request by camelid**
Part of #75080.

@rustbot modify labels: A-intra-doc-links T-doc

### CODE REVIEW

File: library/core/src/future/future.rs [hunk 1]
```diff
@@ -91,11 +91,10 @@ pub trait Future {
...
-    /// [`Poll::Pending`]: ../task/enum.Poll.html#variant.Pending
-    /// [`Poll::Ready(val)`]: ../task/enum.Poll.html#variant.Ready
-    /// [`Context`]: ../task/struct.Context.html
-    /// [`Waker`]: ../task/struct.Waker.html
-    /// [`Waker::wake`]: ../task/struct.Waker.html#method.wake
+    /// [`Poll::Pending`]: Poll::Pending
+    /// [`Poll::Ready(val)`]: Poll::Ready
+    /// [`Waker`]: crate::task::Waker
+    /// [`Waker::wake`]: crate::task::Waker::wake
     #[cfg_attr(not(bootstrap), lang = "poll")]
```
Comment (camelid) [inline library/core/src/future/future.rs:98]: @pietroalbini This line is generating a merge conflict so I'm going to rebase this and have this line be the same as `master`. Is that correct? Also, is there a section in the docs you can point me to on why this is changed at release time?

Comment (pietroalbini) [reply library/core/src/future/future.rs:98]: Some documentation on why `cfg(boostrap)` and `cfg(not(bootstrap))` are needed is [on the rustc-dev-guide](https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#complications-of-bootstrapping). That line should become:

```rust
#[lang = "poll"]
```

**Comment by camelid** [comment]
Rebased.

File: library/core/src/future/future.rs (hunk 1 above)
Comment (camelid) [reply library/core/src/future/future.rs]: So was this a just-stabilized feature then?

File: library/core/src/future/future.rs [hunk 2]
```diff
@@ -91,11 +91,10 @@ pub trait Future {
     /// (memory corruption, incorrect use of `unsafe` functions, or the like),
     /// regardless of the future's state.
     ///
-    /// [`Poll::Pending`]: ../task/enum.Poll.html#variant.Pending
-    /// [`Poll::Ready(val)`]: ../task/enum.Poll.html#variant.Ready
-    /// [`Context`]: ../task/struct.Context.html
-    /// [`Waker`]: ../task/struct.Waker.html
-    /// [`Waker::wake`]: ../task/struct.Waker.html#method.wake
+    /// [`Poll::Pending`]: Poll::Pending
```
Comment (jyn514) [inline library/core/src/future/future.rs:94]:
```suggestion
```

**Review by jyn514 — APPROVED** [review]
r=me with nit fixed

File: library/core/src/future/future.rs (hunk 2 above)
Comment (camelid) [reply library/core/src/future/future.rs:94]: Thanks for catching that!

**Comment by jyn514** [comment]
@bors r+ rollup

@camelid if you find time to squash the commits that would be nice, but it's fine if not.

**Pull request merged**